packages/twenty-server/src/engine/core-modules/auth/token/services/renew-token.service.spec.ts

```diff
@@ -125,6 +131,65 @@ describe('RenewTokenService', () => {
       });
     });
 
+    it('should propagate impersonation claims when present', async () => {
+      const mockRefreshToken = 'valid-refresh-token';
+      const mockUser = { id: 'user-id' } as User;
+      const mockWorkspaceId = 'workspace-id';
+      const mockTokenId = 'token-id';
+      const mockAccessToken = {
+        token: 'new-access-token',
+        expiresAt: new Date(),
+      };
+      const mockNewRefreshToken = {
+        token: 'new-refresh-token',
+        expiresAt: new Date(),
+        targetedTokenType: JwtTokenTypeEnum.ACCESS,
+      };
+      const mockAppToken: Partial<AppToken> = {
+        id: mockTokenId,
+        workspaceId: mockWorkspaceId,
+        user: mockUser,
+        userId: mockUser.id,
+      };
+
+      jest.spyOn(refreshTokenService, 'verifyRefreshToken').mockResolvedValue({
+        user: mockUser,
+        token: mockAppToken as AppToken,
+        authProvider: AuthProviderEnum.Password,
+        targetedTokenType: JwtTokenTypeEnum.ACCESS,
+        isImpersonating: true,
+        impersonationType: 'WORKSPACE' as any,
```

Comment:
??

Reply:
I don't think we should remove impersonationType from the token! I didn't review yet since you were addressing Charles' comment, I can review later today :)